### PR TITLE
Fix knip issues

### PR DESCRIPTION
**Pull request by gitcoindev**
Knip QA

### CODE REVIEW

**Comment by gitcoindev** [comment]
Knip green, QA also done in my forked repository: https://github.com/gitcoindev/permit-generation/pull/2 

I noticed some spelling errors thrown by cspell, give me a minute and I will fix them in one go.

**Comment by gitcoindev** [comment]
Knip green again, spelling issues fixed, pr ready for the review.

**Comment by gitcoindev** [comment]
Interestingly Knip detects `i` in yml wokflow files after yarn command e.g. `yarn i` as a standalone binary with name `i`. I had to change it to `yarn install` and the error was gone. I think we can leave the full `install` for clarity and remember to use it in all GitHub workflow files.

File: knip.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ import type { KnipConfig } from "knip";
 const config: KnipConfig = {
   entry: ["build/index.ts"],
   project: ["src/**/*.ts"],
-  ignore: ["src/types/config.ts"],
+  ignore: ["src/utils/config.ts", "src/utils/helpers.ts", "src/types/github.ts", "src/types/webhook-events.ts"],
```
Comment (gentlementlegen) [inline knip.ts:6]: What is the reason to ignore these files specifically?

Comment (gitcoindev) [reply knip.ts:6]: @gentlementlegen that's a good question and I wanted to discuss this. Knip reports them as unused files and I think they are indeed unused in this repository. I am not sure if they are / should be copied in another repository and I added them to ignore list before this is decided. If you think they can be safely deleted, we can delete the unused files.

Comment (gitcoindev) [reply knip.ts:6]: @gentlementlegen give me a few minutes, I will delete those files and run tests on a clean repo.

Comment (gentlementlegen) [reply knip.ts:6]: Maybe then the way to go is to get this merged, but without ignoring them. This way, we can see from PR https://github.com/ubiquibot/permit-generation/pull/7 if they are indeed unused, in which case I'll remove them safely. Don't see much need to move them to another repo. Worst case we can find get them back with versioning

Comment (gitcoindev) [reply knip.ts:6]: Aha! They are used in tests, will try to add tests to Knip entry points.

Comment (gitcoindev) [reply knip.ts]: @gentlementlegen I added a Jest plugin configuration for Knip, as recommended , see commit https://github.com/ubiquibot/permit-generation/pull/9/commits/6bbe548a9cd0f0b53c2820b46ebbdfcda4939c3a and https://knip.dev/reference/plugins/jest

I also removed removed unused files, removed ignore list and all tests are still passing:

```
$ yarn knip
yarn run v1.22.19
$ knip
✂️  Excellent, Knip found no issues.
Done in 3.39s.
```

and 

```
$ yarn test
yarn run v1.22.19

$ jest
 PASS  tests/register-wallet.test.ts
 PASS  tests/generate-erc721-permit.test.ts

...

Test Suites: 4 passed, 4 total
Tests:       14 passed, 14 total
Snapshots:   0 total
Time:        5.057 s
Ran all test suites.
Done in 6.81s.
```

Comment (gitcoindev) [reply knip.ts]: If you double check that we can indeed remove the files from this repo, please resolve the comment and we can merge .

**Comment by molecula451** [comment]
let me review accordingly  next time lol  @gitcoindev

**Comment by gitcoindev** [comment]
> let me review accordingly next time lol @gitcoindev

Sure, I will!

**Pull request merged**